src/main/java/nl/tudelft/ewi/devhub/server/database/entities/Delivery.java

```diff
@@ -239,6 +234,28 @@ public boolean isLate() {
         return dueDate != null && getTimestamp().after(dueDate);
     }
 
+    @JsonIgnore
+    public void setBuildFailed() {
+        Optional<Mastery> buildFailureMastery = findMasteryByDescription("Build for submitted commit fails");
```

Comment:
My original thought was to create `Mastery` subtypes (or `AutogradedMastery` subtypes) for this purpose, but I can live with this work around for now.